UpdateService/handler.ts

```diff
@@ -0,0 +1,210 @@
+import { identity } from "fp-ts/lib/function";
+import {
+  fromLeft,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import { ServiceModel } from "io-functions-commons/dist/src/models/service";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredParamMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_param";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { Service } from "../generated/api-admin/Service";
+import { SubscriptionKeys } from "../generated/api-admin/SubscriptionKeys";
+import { ServicePayload } from "../generated/definitions/ServicePayload";
+import { ServiceWithSubscriptionKeys } from "../generated/definitions/ServiceWithSubscriptionKeys";
+import { APIClient } from "../utils/clients/admin";
+import { getLogger, ILogger } from "../utils/logging";
+import {
+  ErrorResponses,
+  IResponseErrorUnauthorized,
+  toDefaultResponseErrorInternal,
+  toErrorServerResponse
+} from "../utils/responses";
+import { serviceOwnerCheck } from "../utils/subscription";
+
+type ResponseTypes =
+  | IResponseSuccessJson<ServiceWithSubscriptionKeys>
+  | IResponseErrorUnauthorized
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorNotFound
+  | IResponseErrorTooManyRequests
+  | IResponseErrorInternal;
+
+/**
+ * Type of a UpdateService handler.
+ *
+ * UpdateService expects a service_id and a service payload as input
+ * and returns updated service with subscription keys
+ */
+type IUpdateServiceHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  attrs: IAzureUserAttributes,
+  serviceId: NonEmptyString,
+  servicePayload: ServicePayload
+) => Promise<ResponseTypes>;
+
+const logPrefix = "UpdateServiceHandler";
+
+const getSubscriptionKeysTask = (
+  logger: ILogger,
+  apiClient: ReturnType<APIClient>,
+  serviceId: NonEmptyString
+): TaskEither<ErrorResponses, SubscriptionKeys> =>
+  tryCatch(
+    () =>
+      apiClient.getSubscriptionKeys({
+        service_id: serviceId
+      }),
+    errs => {
+      logger.logUnknown(errs);
+      return toDefaultResponseErrorInternal(errs);
+    }
+  ).foldTaskEither(
+    err => fromLeft(err),
+    errorOrResponse =>
+      errorOrResponse.fold(
+        errs => {
+          logger.logErrors(errs);
+          return fromLeft(toDefaultResponseErrorInternal(errs));
+        },
+        responseType =>
+          responseType.status !== 200
+            ? fromLeft(toErrorServerResponse(responseType))
+            : taskEither.of(responseType.value)
+      )
+  );
+
+const updateServiceTask = (
+  logger: ILogger,
+  apiClient: ReturnType<APIClient>,
+  servicePayload: ServicePayload,
+  serviceId: NonEmptyString
+): TaskEither<ErrorResponses, Service> =>
+  tryCatch(
+    () =>
+      apiClient.updateService({
+        service: {
+          ...servicePayload,
+          authorized_recipients: [],
```

Comment:
Bug here.
From the Developer Portal we can as ADMIN allow one or more fiscal_code, if this way if the user update the service you remove all the fiscal_code we have added.

Reply:
Fixed in [2f63ab6](https://github.com/pagopa/io-functions-services/pull/68/commits/2f63ab68bf93f2db5b11b603e52fa36b04de3306)